rfcs/0002-validation/text.md

```diff
@@ -0,0 +1,149 @@
+# Validation and Conversion of Uploaded Datasets
+
+**Authors:** [Marcus Kinsella](mkinsella@chanzuckerberg.com)
+
+**Approvers:** [Arathi Mani](mailto:arathi.mani@chanzuckerberg.com), [Trent
+Smith](mailto:trent.smith@chanzuckerberg.com), [Brian Raymor](mailto:braymor@chanzuckerberg.com)
+
+## tl;dr
+
+We want datasets in the portal to follow our schema, be viewable in cellxgene, and be available for download in multiple
+file formats. This RFC describes the process and infrastructure for validating uploaded datasets and processing them so
+they are fully available in the portal.
+
+## Problem Statement
+
+There isn't a broad community consensus around how single cell datasets should be stored and annotated. This
+makes it harder to integrate these datasets, which is bad because that's a main aim of our single cell work. So, we have
+a simple schema that we want all datasets to adhere to. If all the datasets we store at least follow this schema, then
+our data corpus will be usable for data integration research and the production of atlases.
+
+While the schema is pretty lightweight, we still need to double-check that uploaded datasets follow it. And, if the
+dataset does not, we want to give the submitter some prompt and useful message explaining why.
+
+Another complication is that single cell datasets come in different formats that are associated with different
+toolchains. We want to support as many of those toolchains as we can, so we're going to convert datasets into
+different formats. It seems to make sense to do this right after validation is complete since we're already parsing
+the original file anyway, and there's not really a good reason to delay converting.
+
+## Product Requirements
+
+The key requirement here is that users can prepare their own datasets without intervention from CZI. This goes along
+with all the other work to support self-submission: file upload, collection creation, and publication.
+
+## Detailed Design
+
+### Overview
+
+![Three Phases of Dataset Creation](imgs/three_phases.png)
+
+There are three main phases in the creation of a new dataset in the data portal:
+
+1. The submitter prepares a dataset file. This is done locally, outside the data portal, assisted by tooling we provide.
```

Comment:
Link in schema/prepare documentation.

Reply:
Will do, though when it eventually gets merged the link will break 😞